transport-classes-io_uring/src/main/java/io/netty5/incubator/channel/uring/AbstractIOUringChannel.java

```diff
@@ -0,0 +1,802 @@
+                                     ReadHandleFactory defaultReadHandleFactory,
+                                     WriteHandleFactory defaultWriteHandleFactory,
+                                     LinuxSocket socket, SocketAddress remote, boolean active) {
+        super(parent, eventLoop, supportingDisconnect, defaultReadHandleFactory, defaultWriteHandleFactory);
+        this.logger = InternalLoggerFactory.getInstance(getClass());
+        this.socket = socket;
+        this.active = active;
+        if (active) {
+            // Directly cache local and remote addresses.
+            local = socket.localAddress();
+            this.remote = remote == null ? socket.remoteAddress() : remote;
+        } else if (remote != null) {
+            this.remote = remote;
+        }
+        if (bufferAllocator().getAllocationType() != StandardAllocationTypes.OFF_HEAP) {
+             setOption(ChannelOption.BUFFER_ALLOCATOR, DefaultBufferAllocators.offHeapAllocator());
+        }
+        prepareClosePromise = eventLoop.newPromise();
+        pendingRead = this::submitReadForPending;
+        rdHubRead = this::submitReadForRdHub;
+        readsPending = new ObjectRing<>();
+        readsCompleted = new ObjectRing<>();
+        cancelledReads = new LongObjectHashMap<>(8);
+    }
+
+    @Override
+    protected SocketAddress localAddress0() {
+        return local;
+    }
+
+    @Override
+    protected SocketAddress remoteAddress0() {
+        return remote;
+    }
+
+    @Override
+    protected void doBind(SocketAddress localAddress) throws Exception {
+        logger.debug("doBind: {} to {}", this, localAddress);
+        if (local instanceof InetSocketAddress) {
+            checkResolvable((InetSocketAddress) local);
+        }
+        socket.bind(localAddress); // Bind immediately, as AbstractChannel expects it to be done after this method call.
+        if (fetchLocalAddress()) {
+            local = socket.localAddress();
+        } else {
+            local = localAddress;
+        }
+        cacheAddresses(local, remoteAddress());
+    }
+
+    protected static void checkResolvable(InetSocketAddress addr) {
+        if (addr.isUnresolved()) {
+            throw new UnresolvedAddressException();
+        }
+    }
+
+    protected final boolean fetchLocalAddress() {
+        return socket.protocolFamily() != SocketProtocolFamily.UNIX;
+    }
+
+    @Override
+    protected void doRead(boolean wasReadPendingAlready) throws Exception {
+        // Schedule a read operation. When completed, we'll get a callback to readComplete.
+        if (submissionQueue == null) {
+            readPendingRegister = true;
+            return;
+        }
+        if (!wasReadPendingAlready) {
+            submitRead();
+        }
+    }
+
+    private void submitRead() {
+        // Submit reads until read handle says stop, we fill the submission queue, or hit max limit
```

Comment:
I need to review this in detail but I think this works quite differently then all there other Channel implementations. Each actual read should be able to influence if a next (from the actual fd) read happens. Here this is not the case as  we always schedule multiple reads.